src/sentry/api/event_search.py

```diff
@@ -1633,31 +1643,68 @@ def parse_function(field, match=None):
     )
 
 
+def count_required_arguments(function):
+    n = 0
+    for arg in function["args"]:
+        if arg.has_default:
+            break
+        n += 1
+    return n
+
+
+def count_total_arguments(function):
+    return len(function["args"])
+
+
+def validate_argument_count(field, function, arguments):
+    """
+    Validate the number of required arguments the function defines against
+    provided arguments. Raise an exception if there is a mismatch in the
+    number of arguments. Do not return any values.
+
+    There are 4 cases:
+        1. provided # of arguments != required # of arguments AND provided # of arguments != total # of arguments (bad, raise an error)
+        2. provided # of arguments < required # of arguments (bad, raise an error)
+        3. provided # of arguments > total # of arguments (bad, raise an error)
+        4. required # of arguments <= provided # of arguments <= total # of arguments (good, pass the validation)
+    """
+    args_count = len(arguments)
+    total_args_count = count_total_arguments(function)
+    if args_count != total_args_count:
+        required_args_count = count_required_arguments(function)
+        if required_args_count == total_args_count:
```

Comment:
I don't think you need this check.

Reply:
This check is in place to give a different error when there aren't any optional arguments. But yes, if it wasn't here, one of the following checks will catch it.